### PR TITLE
FTN-14 Tech Setup

### PR DESCRIPTION
**Pull request by steves-bjss**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -91,6 +91,42 @@ jobs:
...
+                run: dotnet --version
+
+              - name: Restore Packages
+                run: dotnet restore
+
+              - name: Build API
+                run: dotnet build
+
+              - name: Run Unit Tests
+                run: dotnet test
```
Comment (gareth-allan-bjss) [inline .github/workflows/ci.yml:128]: This indentation looks off. I don't think the contents of the `api-build` job need to be indented so far, do they?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -91,6 +91,42 @@ jobs:
...
+            defaults:
+              run:
+                working-directory: ${{ env.api-directory }}
+
+            steps:
+              - name: Checkout source
+                uses: actions/checkout@v4
+
+              - name: Setup dotnet ${{ matrix.dotnet-version }}
+                uses: actions/setup-dotnet@v3
```
Comment (gareth-allan-bjss) [inline .github/workflows/ci.yml:114]: Looks like there's a v4 of this action. Any reason not to use that?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -91,6 +91,42 @@ jobs:
...
+                with:
+                  dotnet-version: ${{ matrix.dotnet-version }}
+
+              - name: Display dotnet version
+                run: dotnet --version
```
Comment (gareth-allan-bjss) [inline .github/workflows/ci.yml:119]: Looks like the "Setup dotnet" step already prints the dotnet version being used, so is this redundant?
Screenshot 2024-12-05 at 16 39 21

File: api/DHSC.FingertipsNext.Modules.Search.ModuleInterfaces/ISearchController.cs
Comment (gareth-allan-bjss) [inline api/DHSC.FingertipsNext.Modules.Search.ModuleInterfaces/ISearchController.cs:1]: Looks like this has been added without removing the `api/DHSC.FingertipsNext.Modules.Search.Interfaces/ISearchController.cs` file, so you've now got two copies of the `ISearchController.cs`.

**Review by gareth-allan-bjss — COMMENTED** [review]
A few comments, but nothing major.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (steves-bjss) [reply .github/workflows/ci.yml]: No they don't - reformatted as suggested.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (steves-bjss) [reply .github/workflows/ci.yml]: No - updated to V4.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (steves-bjss) [reply .github/workflows/ci.yml]: Removed.

File: api/DHSC.FingertipsNext.Modules.Search.ModuleInterfaces/ISearchController.cs
Comment (steves-bjss) [reply api/DHSC.FingertipsNext.Modules.Search.ModuleInterfaces/ISearchController.cs:1]: Removed

**Review by gareth-allan-bjss — APPROVED** [review]

**Pull request merged**